src/main/java/uk/gov/pay/webhooks/webhook/WebhookService.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.pay.webhooks.webhook;
+
+import uk.gov.pay.webhooks.eventtype.EventTypeName;
+import uk.gov.pay.webhooks.eventtype.dao.EventTypeDao;
+import uk.gov.pay.webhooks.eventtype.dao.EventTypeEntity;
+import uk.gov.pay.webhooks.webhook.dao.WebhookDao;
+import uk.gov.pay.webhooks.webhook.dao.entity.WebhookEntity;
+import uk.gov.pay.webhooks.webhook.resource.CreateWebhookRequest;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Optional;
+
+public class WebhookService {
+    WebhookDao webhookDao;
+    
+    EventTypeDao eventTypeDao;
+
+    @Inject
+    public WebhookService(WebhookDao webhookDao, EventTypeDao eventTypeDao) {
+        this.webhookDao = webhookDao;
+        this.eventTypeDao = eventTypeDao;
+    }
+
+    public WebhookEntity createWebhook(CreateWebhookRequest createWebhookRequest) {
+        var webhookEntity = WebhookEntity.from(createWebhookRequest);
+        if (createWebhookRequest.subscriptions() != null) {
+            List<EventTypeEntity> subscribedEventTypes = createWebhookRequest.subscriptions().stream()
+                    .map(eventTypeName -> eventTypeDao.findByName(eventTypeName))
```

Comment:
We might want to avoid a database query in a map in the future but I think given how rarely this will be called and the theoretical limit to the number of entries in the table this shouldn't an issue